server/src/main/java/org/apache/druid/server/coordinator/duty/NewestSegmentFirstIterator.java

```diff
@@ -129,6 +131,12 @@ public class NewestSegmentFirstIterator implements CompactionSegmentIterator
             // For example, if the original is interval of 2020-01-28/2020-02-03 with WEEK granularity
             // and the configuredSegmentGranularity is MONTH, the segment will be split to two segments
             // of 2020-01/2020-02 and 2020-02/2020-03.
+            if (Intervals.ETERNITY.equals(segment.getInterval())
+                && !Granularities.ALL.equals(configuredSegmentGranularity)) {
+              log.warn("Cannot compact segments from ALL to finer granularity for datasource[%s]", dataSource);
```

Comment:
Please include a link to this PR, so future devs can understand why we can't do this. One day, hopefully someone will fix it.

Reply:
https://github.com/apache/druid/issues/13208#issuecomment-1302886876

Reply:
What frequency will this be logged at?

Reply:
Once every `druid.coordinator.period.indexingPeriod` or coordinator custom compaction period, whichever is more frequent

Reply:
Thanks @kfaraz. The custom compaction duty's frequecy is chosen